storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -685,7 +685,7 @@
           0))
       :kill-fn (:report-error-and-die executor-data)
       :factory? true
-      :thread-name component-id)]))
+      :thread-name (str component-id "-executor" (:executor-id executor-data)))]))
```

Comment:
@zhuoliu 
nit: I'd like to add a "-" between "-executor" and executor-id, but it's related to personal style, and really minor.